Mathlib/Data/TypeVec.lean

```diff
@@ -644,12 +729,10 @@ theorem prod_map_id {α β : TypeVec n} : (@TypeVec.id _ α ⊗' @TypeVec.id _ 
 
 @[simp]
 theorem subtypeVal_diagSub {α : TypeVec n} : subtypeVal (repeatEq α) ⊚ diagSub = prod.diag := by
-  ext i x
+  funext i
   induction i with
-  | fz => simp [comp, diagSub, subtypeVal, prod.diag]
-  | fs _ i_ih =>
-    simp only [comp, subtypeVal, diagSub, prod.diag] at *
-    apply i_ih
+  | fz => rfl
+  | fs _ i_ih => apply i_ih
 
 @[simp]
```

Comment:
This looks like an unrelated cleanup of the proof, am I correct?

If this is not actually necessary for your desired change, it's best not to include it in the PR, as it'll just be a distraction and make reviewing it harder (despite it looking like a good cleanup).

This would make for a very nice first extracted PR, though: it's super self-contained and an obvious improvement